client/src/App.tsx

```diff
@@ -1,29 +1,34 @@
 import "./App.css";
 import { useEffect, useState } from "react";
 import AppRouter from "./routers/Router";
-import Navbar from "./components/navbar";
 import ThemeContext, { Theme } from "./context/ThemeContext";
 import UserContext, { UserState } from "./context/UserContext";
 import { User } from "./Interface";
 import Api from "./services/HttpApi";
 
 function App() {
   const [theme, setTheme] = useState("light");
+  const [hasFinishedLoadingUser, setHasFinishedLoadingUser] = useState(false);
```

Comment:
This is necessary because we don't whether the user is authenticated or unauthenticated until `Api.getUser()` is settled.